gobblin-core/src/main/java/gobblin/instrumented/Instrumentable.java

```diff
@@ -33,4 +38,47 @@
    * @return true if instrumentation is enabled, false otherwise.
    */
   public boolean isInstrumentationEnabled();
+
+  /**
+   * Generate tags that should be added to the {@link gobblin.metrics.MetricContext}.
+   * @return List of tags to add to Metric Context.
+   */
+  public List<Tag<?>> generateTags(State state);
+
+  /**
+   * Generate a new metric context replacing old tags with input tags (only tags with the same keys will be replaced),
```

Comment:
Can you use `{@link MetricContext}` and this kind of references for other classes whenever possible?